src/librustc_trans/mir/block.rs

```diff
@@ -412,10 +412,11 @@ impl<'a, 'tcx> FunctionCx<'a, 'tcx> {
                 // is also constant, then we can produce a warning.
                 if const_cond == Some(!expected) {
                     if let Some(err) = const_err {
-                        let err = ConstEvalErr{ span: span, kind: err };
                         let mut diag = bx.tcx().sess.struct_span_warn(
-                            span, "this expression will panic at run-time");
-                        err.note(bx.tcx(), span, "expression", &mut diag);
+                            span, &format!(
+                                "this expression will panic at run-time with {:?}",
+                                err,
+                            ));
```

Comment:
Can we *move* this to const prop? Ideally `rustc_trans` wouldn't emit any warnings!